### ISSUE TITLE
'make' can't write version's string in .pc with releases

### ISSUE THREAD

**Issue by gtm87x** [ccxvii/mujs#45]
Hello,
I'm trying to enable the support for the mpv player's js backend (which uses MuJS) on the brew package manager.
The issue is, the version variable in the MuJS makefile relies on the presence of the .git folder, which is absent on releases' tarballs (used by brew for softwares' stable releases), so it simply doesn't write any version string in the .pc file, making softwares such as mpv that checks specific versions of the library via pkg-config fail to pick it up.
Is there any way to make possible to have the version string written in the .pc in such cases?

**Comment by sebras** [comment]
I believe something along the lines of [this patch](http://git.ghostscript.com/?p=user/sebras/mujs.git;a=commitdiff;h=4656c14d5109c620079f5cd3d5b5c24d15823b35) might resolve the issue.

Not sure if ccxvii agrees with the patch.

Since `git archive` doesn't accept including external files or files that are modified by not committed. I toyed with the idead of adding a `.version` file from which the version is extracted when `.git` is missing. Or modifying `Makefile` when creating the tarball to have `VERSION` not be assigned a  `$(shell ...)` expression but assigned the actual version number. Both do not work because of what files `git archive` include. Of course these operations could be done in a postprocessing step _after_ creating the tarballs, but that is not as neat.

This is why I decided to extract the version from directory name, which definitely is a bit errorprone (what happens if the directory name is not the one we anticipate?).
